src/actor/Verifier.ts

```diff
@@ -0,0 +1,148 @@
+import * as gabi from '@kiltprotocol/portablegabi'
+import CType from '../ctype/CType'
+import {
+  IRequestClaimsForCTypes,
+  MessageBodyType,
+  ISubmitClaimsForCTypes,
+} from '../messaging/Message'
+import AttestedClaim from '../attestedclaim/AttestedClaim'
+import PublicAttesterIdentity from '../attesteridentity/PublicAttesterIdentity'
+
+/**
+ * A helper class to initiate a verification by creating a presentation request which is built
+ * on a specific [[CType]] and attributes of the [[Claim]] the verifier requires to see.
+ */
+export class PresentationRequestBuilder {
+  private builder: gabi.CombinedRequestBuilder
+  private ctypes: Array<CType['hash']>
+  constructor() {
+    this.builder = new gabi.CombinedRequestBuilder()
+    this.ctypes = []
+  }
+
+  /**
+   * Initiates a verification by creating a presentation request for a specific [[CType]].
+   * Note that you are required to call [[finalize]] on the request to conclude it.
+   *
+   * @param p The parameter object.
+   * @param p.ctypehash The SHA-256 hash of the [[CType]].
+   * @param p.attributes A list of attributes of the [[Credential]]s the Verifier has to see in order to verify it.
+   * @param p.legitimations An optional boolean representing whether the Verifier requests to see the legitimations of the Attesters which signed the [[Credential]]s.
+   * @param p.delegationId An optional boolean representing whether the Verifier requests to see the Attesters' unique delegation identifiers.
+   * @param p.reqUpdatedAfter The optional minimum required timestamp on which the [[Credential]] needs to be updated.
+   * The default value for this is the current date.
+   * @returns A [[PresentationRequestBuilder]] on which you need to call [[finalize]] to complete the presentation request.
+   */
+  public requestPresentationForCtype({
+    ctypeHash,
+    attributes,
+    legitimations,
+    delegationId,
+    reqUpdatedAfter = new Date(),
+  }: {
+    attributes: string[]
```

Comment:
Added a type for this in e14866b.

Reply:
I think it is unnecessary to have the type in the `types` folder, since it is only a parameter description for this one function. I would suggest moving it into the file, where it is used.

Reply:
There are already some [other types](https://github.com/KILTprotocol/sdk-js/pull/258/files#diff-d240e310fd6d288f00dbade43d690421) related to the verifier. Should we move them over to? I think it's ok to leave the type there. But maybe not as the default export?

Reply:
Depends on, if these types define a protocol element, which would need to be used, even if you don't use any SDK functions